Pilot/FacebookService.swift

```diff
@@ -29,12 +35,102 @@ class FacebookService: NSObject, FileService {
   internal var basicCredentials: String
 
   /* Default init */
-  required init(preferences: Preferences) {
+  required init(preferences: Preferences, pilotUser: PilotUser) {
     self.preferences = preferences
+    self.pilotUser = pilotUser
 
     basicCredentials = "\(user):\(secret)".dataUsingEncoding(NSUTF8StringEncoding)!.base64EncodedStringWithOptions([])
   }
 
+  // Note: This call is asynronous
+  func refreshCachedCloudContent(completion: ([CloudFile]) -> ()) {
+    // Fetch the photos and videos from facebook and combine them into one conglomerate
+    self.getFacebookPhotos(pilotUser.username, password: pilotUser.password,
+      completion: { returnPhotos in
+        self.getFacebookVideos(self.pilotUser.username, password: self.pilotUser.password,
+          completion: { returnVideos in
+            let returnFiles = returnPhotos + returnVideos
+            self.cachedCloudContent = returnFiles
+            completion(returnFiles)
+          },
+          failure: { _ in
+            ErrorController.sharedErrorController.getMainViewController().errorMessage.stringValue =
```

Comment:
Oh yeah, definitely don't do this. Make the methods on the `ErrorController` object itself, don't want to have to get the `MainViewController` and then edit it directly.